app/uk/gov/hmrc/agentservicesaccount/connectors/AgentClientAuthorisationConnector.scala

```diff
@@ -45,6 +45,7 @@ class AgentClientAuthorisationConnector @Inject()(http: HttpClient)(implicit val
             case OK => Json.parse(response.body).as[SuspensionDetails]
             case NO_CONTENT => SuspensionDetails(suspensionStatus = false, None)
             case NOT_FOUND => throw SuspensionDetailsNotFound("No record found for this agent")
+            case e =>  throw UpstreamErrorResponse(s"Error code $e thrown whilst handing details", e)
```

Comment:
Can you replace the message with "Error $e unable to get suspension details" ? A bit more specific.

Also, for below I was thinking we just log the issue rather than throwing an error for getAgencyDetails eg.

`case s => logger.warn(s"unexpected response $s when getting agency details") None` but it depends, might still want to throw an error there